noir-projects/aztec-nr/aztec/src/utils/bytes/bounded_vec.nr

```diff
@@ -0,0 +1,322 @@
+        assert_eq(output.get(1), 0x202122232425262728292a2b2c2d2e2f303132333435363738393a3b);
+    }
+
+    #[test]
+    fn test_2_fields_to_be_bytes_31() {
+        let input = BoundedVec::<Field, 2>::from_array([
+            0x0102030405060708090a0b0c0d0e0f101112131415161718191a1b1c1d1e1f,
+            0x202122232425262728292a2b2c2d2e2f303132333435363738393a3b,
+        ]);
+        let output: BoundedVec<u8, 62> = fields_to_be_bytes_31(input);
+
+        let mut expected = BoundedVec::new();
+        for i in 1..32 {
+            expected.push(i as u8);
+        }
+        for i in 0..3 {
+            expected.push(0);
+        }
+        for i in 32..60 {
+            expected.push(i as u8);
+        }
+
+        assert_eq(output, expected);
+    }
+
+    #[test]
+    fn test_large_random_be_bytes_31_input_to_fields_and_back(input: [u8; 128]) {
+        let bounded_input = BoundedVec::<u8, 128>::from_array(input);
+        let output = be_bytes_31_to_fields(bounded_input);
+        let input_back = fields_to_be_bytes_31(output);
+
+        assert_eq(bounded_input, input_back);
+    }
+
+    #[test]
+    fn test_large_random_input_to_be_bytes_31_and_back(
+        input1: [u64; 5],
+        input2: [u64; 5],
+        input3: [u64; 5],
+        input4: [u32; 5],
+        input5: [u16; 5],
+        input6: [u8; 5],
+    ) {
+        let mut input = BoundedVec::<Field, 5>::new();
+        for i in 0..5 {
+            let val = (input1[i] as Field * 2.pow_32(184))
+                + (input2[i] as Field * 2.pow_32(120))
+                + (input3[i] as Field * 2.pow_32(56))
+                + (input4[i] as Field * 2.pow_32(24))
+                + (input5[i] as Field * 2.pow_32(8))
+                + input6[i] as Field;
+            input.push(val);
+        }
+
+        let output: BoundedVec<u8, 155> = fields_to_be_bytes_31(input);
+        let input_back = be_bytes_31_to_fields(output);
+
+        assert_eq(input, input_back);
+    }
+
+    #[test(should_fail_with = "call to assert_max_bit_size")]
+    fn test_too_few_destination_be_bytes() {
+        let input = BoundedVec::<Field, 2>::from_array([1, 256]);
+        let _ignored_result: BoundedVec<u8, 32> = fields_to_be_bytes_31(input);
+    }
+
+    #[test(should_fail_with = "call to assert_max_bit_size")]
+    fn test_fields_to_be_bytes_31_value_too_large() {
+        let input = BoundedVec::<Field, 1>::from_array([2.pow_32(248)]);
+        let _ignored_result: BoundedVec<u8, 31> = fields_to_be_bytes_31(input);
+    }
+
+    #[test]
+    fn test_fields_to_be_bytes_31_max_value() {
+        let input = BoundedVec::<Field, 1>::from_array([2.pow_32(248) - 1]);
+        let result: BoundedVec<u8, 31> = fields_to_be_bytes_31(input);
+
+        // Check all bytes are max value (255)
+        for i in 0..31 {
+            assert_eq(result.get(i), 255);
+        }
+    }
+}
+
+mod bounded_vec_specific {
```

Comment:
These tests are unique to the BoundedVec - I test here the case where the input vecs are not fully populated.